src/Core/DocumentUrl.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * BigBlueButton open source conferencing system - https://www.bigbluebutton.org/.
+ *
+ * Copyright (c) 2016-2024 BigBlueButton Inc. and by respective authors (see below).
+ *
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation; either version 3.0 of the License, or (at your option) any later
+ * version.
+ *
+ * BigBlueButton is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with BigBlueButton; if not, see <https://www.gnu.org/licenses/>.
+ */
+
+namespace BigBlueButton\Core;
+
+class DocumentUrl extends Document
+{
+    private string $url;
+
+    public function __construct(string $url, ?string $name = null)
+    {
+        $this->setUrl($url);
+        $this->setName($name);
+    }
+
+    public function setUrl(string $url): self
+    {
+        $this->url = $url;
+
+        return $this;
+    }
+
+    public function getUrl(): string
+    {
+        return $this->url;
+    }
+
+    public function isValid(): bool
+    {
+        return $this->urlExists($this->getUrl());
+    }
+
+    private function urlExists(string $url): bool
```

Comment:
This function must stay very optional, Timeout can be problematic, if we have 4 files to pre-upload, we can have 20 seconds delay using the function and can lead to user frustration. BigBlueButton already handles it for us.

It might be useful for some use cases but too dangerous, should be marked as `@experimental` at least.

Reply:
function marked

```php
    /**
     * Checks the validity / existence of the provided URL. Pending on file size and server-performance
     * the response could be slow.
     *
     * @experimental
     */
    private function urlExists(string $url): bool
```